pkg/api/v1beta1/types.go

```diff
@@ -1081,6 +1081,17 @@ type ListOptions struct {
 	ResourceVersion string `json:"resourceVersion" description:"when specified with a watch call, shows changes that occur after that particular version of a resource; defaults to changes from the beginning of history"`
 }
 
+// PodLogOptions is the query options for a Pod's logs REST call
+type PodLogOptions struct {
+	TypeMeta `json:",inline"`
+
+	// Container for which to return logs
+	Container string `json:"container" description:"the container for which to stream logs; defaults to first container in pod"`
```

Comment:
Config merge is not going to be order-preserving: #6027. The list of containers effectively needs to be treated as an unordered map. Defaulting to the only container if there is only 1 would be fine. The "first" isn't well defined.

Also, optional fields must be tagged with `,omitempty`.